### DISCUSSION TITLE
关于底层传输方式增加WebTransport的想法

### DISCUSSION THREAD

**Discussion by songee2020** [v2fly/v2ray-core#1712] [Ideas]
对于v2ray支持quic或者http3的问题，有不少人关注，结合目前的信息，分享一下个人的业余观点。
目前HTTP/3提案 [RFC 9114](https://datatracker.ietf.org/doc/rfc9114/) 已定稿，上游项目 [lucas-clemente/quic-go](https://github.com/lucas-clemente/quic-go) 也已支持，需要测试的可以参考  [add VMess-QUIC](https://github.com/XTLS/Xray-examples/pull/51/files) 该方案基于现有的底层传输方式QUIC。这种方式的优点在于可以直接使用HTTP/3协议而不必修改v2ray代码，然而却不能像WebSocket那样支持路径分流。为了做到更好的伪装，避免主动探测，有必要增加WebTransport这种底层传输方式。
对WebTransport的介绍，参考 [Experimenting with QUIC and WebTransport in Go](https://centrifugal.github.io/centrifugo/blog/quic_web_transport/) 和 [尝试使用 WebTransport](https://web.dev/webtransport/) 这两篇文章。[WebTransport over HTTP/3](https://datatracker.ietf.org/doc/draft-ietf-webtrans-http3/) 可以实现的功能类似于 [WebSocket](https://datatracker.ietf.org/doc/html/rfc6455) over HTTP/1.1，也就是很多人在用的Vmess+wss+Web方案。根据 [RFC 6455](https://datatracker.ietf.org/doc/html/rfc6455) 文档说明“The WebSocket Protocol is an independent TCP-based protocol”，基于TCP协议的 [WebSocket](https://github.com/gorilla/websocket) 并不适合HTTP3，只能使用WebTransport。
希望社区的maintainers能够关注这个方向。

**Comment by AkinoKaede** [comment]
Thanks. Actually, I am develping this feature, but the development of [github.com/marten-seemann/webtransport-go](https://github.com/marten-seemann/webtransport-go) has not been completed.

**Comment by escape0707** [comment]
See also: [RFC 9220: Bootstrapping WebSockets with HTTP/3](https://www.rfc-editor.org/rfc/rfc9220.html)
